modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -328,106 +337,289 @@ def find_esc3_vuln_cert_templates
+      password: pass,
+      transport: :negotiate
+    )
 
-    return if esc_entries.empty?
+    # Check if registry values are already cached
+    cached_values = @ldap_objects.find { |obj| obj[:type] == :registry_values }
+    return cached_values[:values] if cached_values
 
-    # Determine if the user we've authenticated with has the ability to edit
-    esc_entries.each do |entry|
-      certificate_symbol = entry[:cn][0].to_sym
-      next if @certificate_details[certificate_symbol][:enroll_sids].empty?
+    registry_values = {}
+
+    begin
+      conn.shell(:powershell) do |shell|
+        registry_values[:certificate_mapping_methods] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Control\\SecurityProviders\\Schannel', 'CertificateMappingMethods')
+        registry_values[:strong_certificate_binding_enforcement] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\Kdc', 'StrongCertificateBindingEnforcement')
+
+        active_policy_name = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'Active')
+        registry_values[:disabled_extension_list] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'DisableExtensionList', active_policy_name)
+        registry_values[:edit_flags] = run_registry_command(shell, 'HKLM:\\SYSTEM\\CurrentControlSet\\Services\\CertSvc\\Configuration\\*\\PolicyModules', 'EditFlags', active_policy_name)
+      end
+
+      if registry_values[:strong_certificate_binding_enforcement] == '1'
+        vprint_good('ESC9 could be exploitable using Kerberos due to StrongCertificateBindingEnforcement being set to 1')
+      elsif registry_values[:strong_certificate_binding_enforcement] == '0'
+        vprint_good('ESC10 could be exploitable using Kerberos due to StrongCertificateBindingEnforcement being set to 0')
+      elsif registry_values[:strong_certificate_binding_enforcement] == '2'
+        vprint_error('ESC9 and ESC10 cannot be exploited using Kerberos due to StrongCertificateBindingEnforcement being set to 2')
+      end
+
+      if registry_values[:certificate_mapping_methods] == '4' || registry_values[:certificate_mapping_methods] == '31'
+        vprint_good('ESC10 could be exploitable using Schannel due to CertificateMappingMethods allowing UPN certificate mapping')
+      else
+        print_error('ESC10 cannot be exploited using Schannel due to CertificateMappingMethods not allowing UPN certificate mapping')
+      end
+
+      vprint_status("Registry Values Retrieved for ESC9 & ESC10: #{registry_values}")
+    rescue StandardError => e
+      vprint_warning("Failed to query registry values for ESC9 and ESC10 - this is expected when running as a low-privilege users. Error: #{e.message}")
+    end
+
+    # Cache the registry values in @ldap_objects
+    @ldap_objects << { type: :registry_values, values: registry_values }
```

Comment:
This just doesn't exactly seem right. The `@ldap_objects` should just contain ldap objects, caching registry values and the authenticated user is great but they should be kept separate because it's confusing to store things that aren't LDAP object here. Maybe `@registry_values`, keyed by the registry key would be a better fit?